subprojects/core/src/main/java/org/gradle/api/internal/tasks/execution/SkipCachedTaskExecuter.java

```diff
@@ -94,22 +99,30 @@ public void readFrom(InputStream input) throws IOException {
                             });
                             if (found) {
                                 state.upToDate("FROM-CACHE");
+                                taskOutputCacheListener.fromCache(task);
                                 return;
                             }
+                            reason = NotCachedReason.NOT_IN_CACHE;
                         } catch (Exception e) {
                             LOGGER.warn("Could not load cached output for {} with cache key {}", task, cacheKey, e);
+                            reason = NotCachedReason.ERROR_LOADING_CACHE_ENTRY;
                         }
                     }
                 } else {
                     LOGGER.info("Not caching {} because it declares multiple output files for a single output property via `@OutputFiles`, `@OutputDirectories` or `TaskOutputs.files()`", task);
+                    reason = NotCachedReason.MULTIPLE_OUTPUTS;
```

Comment:
If we want to have logging done by a listener then we need to capture this as a different reason.